orttraining/orttraining/training_api/optimizer.cc

```diff
@@ -63,19 +64,25 @@ std::unique_ptr<OptimizerAlgorithmBase> OptimizerAlorithmFactory::CreateInstance
     const std::string& optim_path_or_bytes, int32_t& group_count) {
   std::map<std::pair<std::string, std::string>, int32_t> opt_type_to_freq_map;
 #if !defined(ORT_MINIMAL_BUILD)
-  std::shared_ptr<Model> model;
-  ORT_ENFORCE(Model::Load(ToWideString(optim_path_or_bytes), model, nullptr,
-                          logging::LoggingManager::DefaultLogger())
-                  .IsOK());
-  Graph& graph = model->MainGraph();
-  for (auto& node : graph.Nodes()) {
-    if (node.Domain() == kMSDomain && (node.OpType() == "AdamWOptimizer" || node.OpType() == "SGDOptimizerV2")) {
-      auto domain_type_pair = std::make_pair(node.Domain(), node.OpType());
-      if (opt_type_to_freq_map.find(domain_type_pair) == opt_type_to_freq_map.end()) {
-        opt_type_to_freq_map[domain_type_pair] = 0;
-      }
+  if (fbs::utils::IsOrtFormatModel(ToWideString(optim_path_or_bytes))) {
```

Comment:
we can call `ToWideString()` (or preferably the more descriptive `ToPathString()`) just once:
```c++
if (const auto optim_path_str = ToPathString(optim_path_or_bytes);
    fbs::utils::IsOrtFormatModel(optim_path_str)) {
  ...
} else {
  ...
}
```

also `optim_path_or_bytes` implies that we would handle model bytes here too, but that doesn't look like the case.